src/components/LoginPage.js

```diff
@@ -0,0 +1,164 @@
+                  Username
+                </label>
+                <input
+                  id="username"
+                  type="text"
+                  placeholder="Username"
+                  value={username}
+                  onChange={(e) => setUsername(e.target.value)}
+                  required
+                  className="shadow appearance-none border rounded w-full py-2 px-3 text-gray-700 leading-tight focus:outline-none focus:ring-2 focus:ring-blue-500"
+                />
+              </div>
+            )}
+            <div className="mb-4">
+              <label className="block text-gray-700 text-sm font-bold mb-2" htmlFor="email">
+                Email
+              </label>
+              <input
+                id="email"
+                type="email"
+                placeholder="Email"
+                value={email}
+                onChange={(e) => setEmail(e.target.value)}
+                required
+                className="shadow appearance-none border rounded w-full py-2 px-3 text-gray-700 leading-tight focus:outline-none focus:ring-2 focus:ring-blue-500"
+              />
+            </div>
+            <div className="mb-4">
+              <label className="block text-gray-700 text-sm font-bold mb-2" htmlFor="password">
+                Password
+              </label>
+              <input
+                id="password"
+                type="password"
+                placeholder="Password"
+                value={password}
+                onChange={(e) => setPassword(e.target.value)}
+                required
+                className="shadow appearance-none border rounded w-full py-2 px-3 text-gray-700 leading-tight focus:outline-none focus:ring-2 focus:ring-blue-500"
+              />
+            </div>
+            {error && <p className="text-red-500 text-xs italic mb-4 text-center">{error}</p>}
+            {successMessage && <p className="text-green-500 text-xs italic mb-4 text-center">{successMessage}</p>}
+            <div className="flex items-center justify-between">
+              <button
+                type="submit"
+                className="w-full bg-blue-500 hover:bg-blue-700 text-white font-bold py-2 px-4 rounded focus:outline-none focus:ring-2 focus:ring-blue-500"
+              >
+                {isSignUp ? 'Sign Up' : 'Sign In'}
+              </button>
+            </div>
+          </form>
+          <div className="text-center mt-6">
+            <button
+              onClick={() => setIsSignUp(!isSignUp)}
+              className="text-blue-500 hover:text-blue-700 font-bold transition duration-300"
+            >
+              {isSignUp ? 'Already have an account? Sign In' : 'Don’t have an account? Sign Up'}
+            </button>
+          </div>
+          <div className="text-center mt-6 flex justify-center space-x-4">
+            <a href="http://localhost:3000/auth/google" className="text-gray-700 hover:text-red-600 transition duration-300">
```

Comment:
i see that u are able to use that signup at your local host right?

Reply:
yeah lets make it environment independent